### PR TITLE
UT of matrix.print(std::ostream &os,const double threshold);

### PR DESCRIPTION
**Pull request by KCJ2000**
fix#2058

### CODE REVIEW

**Comment by hongriTianqi** [comment]
> fix#2058

should be fix #2058

File: source/module_base/matrix.cpp [hunk 1]
```diff
@@ -321,33 +323,34 @@ double matrix::trace_on(void) const
...
-
-    for (int i = 1; i < nr; ++i)
-    {
-        if (a[i] < ev_lower) ev_lower = a[i];
-        if (b[i] > ev_upper) ev_upper = b[i];
-    }
-    delete[] a;
-    delete[] b;
-}
+//this function is wrong, and there is no file use this function.
```
Comment (hongriTianqi) [inline source/module_base/matrix.cpp:326]: Why is it wrong?

Comment (KCJ2000) [reply source/module_base/matrix.cpp:326]: The name of the function is void matrix::get_extreme_eigen_values,which means that this function should return the maximum and minimum of the eigenvalues of this matrix. However, firstly it has no insurance code to make sure the matrix is a square matrix. Secondly, there is no code to solve the eigenvalues of the matrix. This function is right if and only if the matrix has been diagonalized. The same problem is that there is no insurance code and it's not the most efficient algorithm.
In conclusion, this function can't perfectly finish the aim defined by its name. And it has not been called.
https://hongritianqi.github.io/abacus-develop/source/html/classModuleBase_1_1matrix.html#a976ec94fbdf0a34ae89c92cdcfb9ddc8

**Review by hongriTianqi — APPROVED** [review]
